### PR TITLE
[JENKINS-70021] fix tests when gpg signing is enabled

### PR DESCRIPTION
**Pull request by olamy**
## [JENKINS-70021](https://issues.jenkins.io/browse/JENKINS-70021) - fix tests when gpg signing is enabled

if git global config gpg-sign activated e.g git config --global commit.gpgsign true.

unit test of git-plugin fail 

cli 
```
Caused by: hudson.plugins.git.GitException: Command "git commit -F /Volumes/workspace/dev/sources/jenkins/git-plugin/target/tmp/junit11293633318696642182/junit5698144647230035770@tmp/jenkins-gitclient-gitcommit4252944134817268881.txt" returned status code 128:stdout: stderr: error: gpg failed to sign the datafatal: failed to write commit object
{{ at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2671)}}{{ at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2601)}}
```

jgit

```
{}Caused by: org.eclipse.jgit.api.errors.ServiceUnavailableException: Signing service is not available{{ at org.eclipse.jgit.api.CommitCommand.sign(CommitCommand.java:290)}}{{ at org.eclipse.jgit.api.CommitCommand.call(CommitCommand.java:247)}}
```

same when using GitSampleRepoRule

```
{}Initialized empty Git repository in /Volumes/workspace/dev/sources/jenkins/cloudbees-analytics-plugin/plugin/target/tmp/junit3925781698945114105/junit839357165607123398/.git/[junit839357165607123398] $ git add file[junit839357165607123398] $ git config user.name "Git SampleRepoRule"[junit839357165607123398] $ git config user.email gits@mplereporule[junit839357165607123398] $ git commit --message=initerror: gpg failed to sign the datafatal: failed to write commit object   1.341 [id=802] INFO hudson.lifecycle.Lifecycle#onStatusUpdate: Stopping Jenkins   1.511 [id=802] INFO hudson.lifecycle.Lifecycle#onStatusUpdate: Jenkins stopped
```
 

there are different options to use:

cli env var GIT_CONFIG_NOSYSTEM
jgit SystemReader.getInstance().getUserConfig().clear();

to reproduce the failures before those changes. 
Just activate gpgsigning
```
git config --global commit.gpgsign true
```

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-plugin/blob/master/CONTRIBUTING.adoc) doc
- [x] I have referenced the Jira issue related to my changes in one or more commit messages
- [x] I have added tests that verify my changes
- [x] Unit tests pass locally with my changes
- [x] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [x] Documentation in README has been updated as necessary
- [x] Online help has been added and reviewed for any new or modified fields
- [x] I have interactively tested my changes
- [x] Any dependent changes have been merged and published in upstream modules (like git-client-plugin)

## Types of changes

- [x] Dependency or infrastructure update

### CODE REVIEW

File: src/test/java/hudson/plugins/git/TestGitRepo.java [hunk 1]
```diff
@@ -34,9 +35,9 @@ public class TestGitRepo {
     public TestGitRepo(String name, File tmpDir, TaskListener listener) throws IOException, InterruptedException {
 		this.name = name;
 		this.listener = listener;
-		
-		EnvVars envVars = new EnvVars();
-		
+
+        EnvVars envVars = GitUtilsTest.getConfigNoSystemEnvsVars();
```
Comment (olamy) [inline src/test/java/hudson/plugins/git/TestGitRepo.java:39]: well the code here use tabs whereas I'm using spaces :)

**Review by MarkEWaite — APPROVED** [review]

File: src/test/java/hudson/plugins/git/TestGitRepo.java (hunk 1 above)
Comment (MarkEWaite) [reply src/test/java/hudson/plugins/git/TestGitRepo.java]: I adjusted it to reduce white space diffs as mentioned in the contributing guide.  Thanks very much for the fixes!

Comment (olamy) [reply src/test/java/hudson/plugins/git/TestGitRepo.java]: thanks.
fair enough. but it's not really convenient to have files using tabs and files using spaces within the same project.
Intellij doesn't have a per file configuration feature :)

**Comment by olamy** [comment]
any chance to release it soon?

**Comment by MarkEWaite** [comment]
> any chance to release it soon? thanks

Likely before the end of this week.

**Pull request merged**